packages/client/package.json

```diff
@@ -19,9 +19,41 @@
     "typescript",
     "query-builder"
   ],
-  "main": "index.js",
-  "browser": "index-browser.js",
   "types": "index.d.ts",
+  "main": "index.js",
+  "module": "index.mjs",
+  "browser": "index-browser.mjs",
+  "exports": {
+    "./package.json": "./package.json",
+    ".": {
+      "types": "./index.d.ts",
```

Comment:
It's not 100% correct to use `types` condition like this. And in this specific case, it's even more apparent. Notice how your `index.d.ts` **doesn't** match `index.mjs`:
```ts
// index.d.ts
export * from '.prisma/client'

// index.mjs
export * from '../../.prisma/client/index.mjs'
export * as default from '../../.prisma/client/index.mjs'
```
It's clearly visible that this declaration file doesn't describe closely this runtime file and yet you are saying (through the `types` condition applied like this) that it does.

Reply:
Ok, then this is just half of the solution? How can we reconciliate both requirements: that an import from `@prisma/client` points to the right types (this is what this is doing, taken from ts [docs](https://www.typescriptlang.org/docs/handbook/esm-node.html#packagejson-exports-imports-and-self-referencing)), but also allow types to be matched on fully specified imports. I think this needs to be handled in the `*` pattern below. Though I haven't tried yet.

Reply:
> How can we reconciliate both requirements: that an import from @prisma/client points to the right types (this is what this is doing, taken from ts [docs](https://www.typescriptlang.org/docs/handbook/esm-node.html#packagejson-exports-imports-and-self-referencing)),

That is not exactly what this PR is doing - although at some point the TS docs were showing what you have in this PR, that was a mistake and the docs were changed. Notice that now the docs show an example with smth like this:
```json
{
    "exports": {
        ".": {
            // Entry-point for `import "my-package"` in ESM
            "import": {
                // Where TypeScript will look.
                "types": "./types/esm/index.d.ts",
                // Where Node.js will look.
                "default": "./esm/index.js"
            },
            // Entry-point for `require("my-package")` in CJS
            "require": {
                // Where TypeScript will look.
                "types": "./types/commonjs/index.d.cts",
                // Where Node.js will look.
                "default": "./commonjs/index.cjs"
            },
        }
    },
}
```

We can see here that the types are kinda duplicated between import and require conditions - and that matches exactly what you are doing at runtime. You are providing 2 totally different files for the runtime so it kinda only makes sense to do the same for types (I recognize that this is quite annoying but that's what is *correct*).

> but also allow types to be matched on fully specified imports (what you are showing). I think this needs to be handled in the * pattern below, and this is not so much of an issue actually? Though I haven't tried yet.

It would be easier for me to discuss this using some examples and stuff.

Reply:
Do the files actually differ in content? These entrypoint files/types redirect to our actual generated types. We want to keep the experience the same between ESM/CJS, so we will have the same default exports, and named exports, APIs, etc...